databuilder/databuilder/task/search/document_mappings.py

```diff
@@ -63,68 +65,94 @@ class SearchableResource(Document):
     # https://www.elastic.co/guide/en/elasticsearch/reference/current/keyword.html#wildcard-field-type
     key = Text(required=True,
                fields={"keyword": Keyword()},
-               analyzer=Analyzer.general_analyzer)
+               analyzer=Analyzer.general_analyzer,
+               term_vector=POSITIONS_OFFSETS)
     name = Text(required=True,
                 fields={"keyword": Keyword()},
-                analyzer=Analyzer.stemming_analyzer)
+                analyzer=Analyzer.stemming_analyzer,
+                term_vector=POSITIONS_OFFSETS)
     description = Text(analyzer=Analyzer.english_analyzer,
-                       fields={"alphanumeric": Text(analyzer=Analyzer.alphanum_analyzer)})
+                       fields={"alphanumeric": Text(analyzer=Analyzer.alphanum_analyzer,
+                                                    term_vector=POSITIONS_OFFSETS)
+                               },
+                       term_vector=POSITIONS_OFFSETS)
     badges = Text(multi=True,
                   fields={"keyword": Keyword()},
-                  analyzer=Analyzer.general_analyzer)
+                  analyzer=Analyzer.general_analyzer,
+                  term_vector=POSITIONS_OFFSETS)
     tags = Text(multi=True,
                 fields={"keyword": Keyword()},
-                analyzer=Analyzer.general_analyzer)
-    usage = RankFeatures()  # values can be used to boost document search score
+                analyzer=Analyzer.general_analyzer,
+                term_vector=POSITIONS_OFFSETS)
+    usage = RankFeatures()
     last_updated_timestamp = Date()
-
     resource_type = Keyword(required=True)
 
 
 class Table(SearchableResource):
+    columns = Text(multi=True,
+                   fields={
+                       "keyword": Keyword(),
+                       "general": Text(multi=True,
+                                       analyzer=Analyzer.general_analyzer,
+                                       term_vector=POSITIONS_OFFSETS)
+                   },
+                   term_vector=POSITIONS_OFFSETS,
```

Comment:
what does it mean to have POSITIONS_OFFSETS here and in `general`

Reply:
are there actually 3 fields?
The toplevel `columns` of type Text, and then `keyword` and `general` subfields. With `general` being a Text as well, but with a different analyzer.

Reply:
There are 2 different analyzers for the fields and its subfield and since the term vectors contain data about how the field was analyzed it would be different on both.